src/rgw/rgw_sal_motr.cc

```diff
@@ -1933,8 +1933,19 @@ int MotrObject::MotrReadOp::prepare(optional_yield y, const DoutPrefixProvider*
   if (rc < 0)
     return rc;
 
-  if (ent.is_delete_marker())
-    return -ENOENT;
+  req_state* s = static_cast<req_state*>(rctx->get_private());
+
+  // In GET/HEAD object API, return "MethodNotAllowed"
+  // if delete-marker is the latest object entry
+  // Else, return "NoSuchKey" error
+  if (ent.is_delete_marker()) {
+    if (source->get_instance() == ent.key.instance && ent.key.instance != "") {
+      s->err.message = "The specified method is not allowed against this resource.";
```

Comment:
I believe a debug log entry stating why `MethodNotAllowed` error occurred would be helpful.

Reply:
agree. 
infact error message also needs to be specific, something like
"   s->err.message = "The <GET/HEAD whatever applies from request> method not allowed against delete marker <version-id>";"

Reply:
Yes, added the debug log.

Reply:
```
`infact error message also needs to be specific, something like
" s->err.message = "The <GET/HEAD whatever applies from request> method not allowed against delete marker ";"`
```
Currently, MGW returns "Unknown" in the error message. To change it and make it inline with AWS, I have added the "The specified method is not allowed against this resource." error message. Now MGW is inline with AWS.